contracts/contracts/coordination/AllowList.sol

```diff
@@ -31,23 +39,27 @@ contract AllowList is AccessControlDefaultAdminRules, IRitualAuthorizer, RitualS
         return authorizations[ritualId][enricoAddress];
     }
 
-    function authorize(uint32 ritualId, address[] calldata addresses) public {
+    function authorize(uint32 ritualId, address[] calldata addresses) public payable {
+        require(msg.value > 10000, "Insufficient fee");
         require(coordinator.getAuthority(ritualId) == msg.sender,
             "Only ritual authority is permitted");
         require(coordinator.getRitualState(ritualId) == RitualState.FINALIZED,
             "Only active rituals can add authorizations");
         for (uint i=0; i<addresses.length; i++) {
             authorizations[ritualId][addresses[i]] = true;
         }
+        feeRecipient.transfer(msg.value);
     }
 
-    function deauthorize(uint32 ritualId, address[] calldata addresses) public {
+    function deauthorize(uint32 ritualId, address[] calldata addresses) public payable {
+        require(msg.value > 10000, "Insufficient fee");
```

Comment:
Hmm, why send ether to this function?

Reply:
This is an example of a payment gateway. If an adopter wants to edit the authorization they need to pay